client/my-sites/checkout/composite-checkout/components/item-variation-picker/variant-dropdown-price.tsx

```diff
@@ -244,10 +245,12 @@ export const ItemVariantDropDownPrice: FunctionComponent< {
 				) }
 			</Label>
 			<PriceTextContainer>
-				{ discountPercentage > 0 && ! isJetpack && ! isMobile && ! isIntroductoryOffer && (
-					<DiscountPercentage percent={ discountPercentage } />
-				) }
-				{ discountPercentage > 0 && ! isJetpack && ! isIntroductoryOffer && (
+				{ discountPercentage > 0 &&
+					! isMobile &&
+					( ! isIntroductoryOffer || shouldRenderJetpackDiscount ) && (
```

Comment:
Is there any way we can clear up this logic a bit? I think I understand what we are checking but I think it would be nice to save this to some variable with a comment above explaining it clearly

Reply:
I'll try to. Can't promise anything. :D 
Due to the nature of each condition, it's a bit hard to simplify the logic here.